### ISSUE TITLE
Create Start Menu folder option only creates an empty file

### ISSUE THREAD

**Issue by BC46** [oliverpechey/Freelancer-HD-Edition-Installer#79]
Whenever the installation program creates a Start Menu folder, it just adds an empty file to it, nothing else. Obviously, the Start Menu shortcut doesn't work because of this.

![image](https://user-images.githubusercontent.com/45081813/160689589-9c9963e1-ad35-40ad-ac7c-6ffef374fbee.png)

**Comment by ollieraikkonen** [comment]
ChangesAssociations=yes might resolve this. So #80 might close this.

**Comment by BC46** [comment]
Yep, should be automatically closed after merging.

**Comment by BC46** [comment]
Unfortunately this issue still persists with `ChangesAssociations=yes` and the all-in-one installer. The problem is that the shortcut is created before Freelancer.exe gets extracted.

**Comment by oliverpechey** [comment]
I wonder if you can programmatically call a desktop refresh via the Win API?

**Comment by oliverpechey** [comment]
@BC46 https://stackoverflow.com/questions/5849289/inno-setup-refresh-desktop

**Comment by oliverpechey** [comment]
Actually it might just be easier to add an [Icon] section. See top answer: https://stackoverflow.com/questions/32262254/how-to-create-a-desktop-icon-with-inno-setup

This allows you to set an icon rather than using the one from the exe (which doesnt exist yet)

`[Icons]
Name: "{userdesktop}\{#MyAppName}"; Filename: "{app}\{#MyAppExeName}"; \
    IconFilename: "{app}\{#MyAppIcoName}"; Tasks: desktopicon`

**Comment by BC46** [comment]
Turns out the issue had nothing to do with the order of how things were done during the installation, but rather with the name of the icon.

We currently have:
```iss
Name: "{group}\{#MyAppName}"; Filename: "{app}\EXE\{#MyAppExeName}"
```

`{#MyAppName}` expands to `Freelancer: HD Edition v0.6`, which is an invalid file name on Windows due to the colon. The issue can be fixed by simply changing the name to `Freelancer HD Edition`:
```iss
Name: "{commondesktop}\Freelancer HD Edition"; Filename: "{app}\EXE\{#MyAppExeName}"; Tasks: desktopicon
```

I also changed `group` to `commondesktop` because this way only a shortcut will be added to the start menu, instead of a whole folder. `Freelancer HD Edition` should be also added as a constant called "MyValidAppName" or something, because this string is used in other places as well.

However, with the applied fix I noticed one other small issue, which was that the start menu showed a blank icon for the HDE executable, even after the icon refresh post-install. The fix I found for this was to copy the `icon.ico` file to the location of Freelancer.exe before the actual exe is moved there:
```iss
Source: "Assets\Images\icon.ico"; DestDir: "{app}\EXE"; DestName: "{#MyAppExeName}"; Flags: ignoreversion
```
This makes the start menu actually believe that this is the correct icon, and thus it will instantly display it. Obviously this is not a clean solution but it gets the job done.
